ksiren-okhttp3-plugin/src/main/java/com/brightspace/ksiren/okhttp3_plugin/KSirenOkhttp3EntityFetchClient.kt

```diff
@@ -0,0 +1,25 @@
+package com.brightspace.ksiren.okhttp3_plugin
+
+import com.brightspace.ksiren.KSirenEntityFetchClient
+import com.brightspace.ksiren.KSirenJsonReader
+import com.brightspace.ksiren.KSirenJsonReaderFactory
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import java.io.IOException
+
+/**
+ * Created by mpomeroy on 11/2/17.
+ */
+class KSirenOkhttp3EntityFetchClient(val okhttpClient: OkHttpClient, val ksirenJsonReaderFactory: KSirenJsonReaderFactory): KSirenEntityFetchClient() {
+	override fun executeCall(href: String): KSirenJsonReader {
+		return ksirenJsonReaderFactory
+				.getKSirenJsonReader(okhttpClient.newCall(
+					Request
+						.Builder()
+						.url(href)
+						.build())
+					.execute()
```

Comment:
Won't this block the thread? Are you switching to a background thread before this call and then going back to the main thread afterwards?

Reply:
Well, it should throw a network on main exception, if anything. I don't think threading it in a way that works for our use case is the responsibility of KSiren, though. 

Okhttp3, for instance, doesn't prevent you, at a code level, from running a request on the main thread, it just throws an exception if you try it.